Infusionsoft/App.php

```diff
@@ -196,8 +230,16 @@ public function hasTokens(){
     }
 
     public function refreshTokens(){
-        $tokens = Infusionsoft_OAuth2::refreshToken($this->refreshToken);
-        $this->updateAndSaveTokens($tokens['access_token'], $tokens['refresh_token'], $tokens['expires_in']);
+        if (method_exists($this->tokenStorageProvider, 'refreshTokens')){

```

Comment:
This seems fairly hokey.  A token storage provider shouldn't refresh the tokens.